### PR TITLE
DXCDT-295: Refactor quickstarts command to use quickstart meta URL

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

Given that all the information we need to fetch a quickstart and download it is available at https://auth0.com/docs/meta/quickstarts, we no longer need to bake in the paths and meta info inside the cli, but instead retrieve it from the url.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/auth0/quickstart.go [hunk 1]
```diff
@@ -1,10 +1,150 @@
...
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("expected status %d, got %d", http.StatusOK, response.StatusCode)
+	}
+
+	tmpFile, err := os.CreateTemp("", "auth0-quickstart*.zip")
```
Comment (Widcket) [inline internal/auth0/quickstart.go:73]: Is the `*` in the name intentional?

File: internal/cli/quickstarts.go [hunk 2]
```diff
@@ -111,218 +96,112 @@ func downloadQuickstartCmd(cli *cli) *cobra.Command {
...
+		if err != nil {
+			return fmt.Errorf("failed to compute the path where to download the quickstart sample: %w", err)
+		}
 
-	request.URL.RawQuery = params.Encode()
-	request.Header.Set("Content-Type", quickstartContentType)
+		if pathExists && !cli.force {
+			message := fmt.Sprintf(
+				"%s %s already exists.\n Directory contents will be replaced. Are you sure you want to proceed? ",
+				ansi.Yellow("WARNING:"),
```
Comment (Widcket) [inline internal/cli/quickstarts.go:137]: Do we have a renderer function for warnings?

File: internal/cli/quickstarts.go [hunk 3]
```diff
@@ -465,3 +344,54 @@ func defaultCallbackURLFor(s string) string {
 func defaultURL(url string, port int) string {
 	return fmt.Sprintf("%s:%d", url, port)
 }
+
+func (i *qsInputs) fromArgs(cmd *cobra.Command, args []string, cli *cli) error {
```
Comment (Widcket) [inline internal/cli/quickstarts.go:348]: This is useful, could be an interface method in the future.

File: internal/cli/quickstarts.go [hunk 4]
```diff
@@ -465,3 +344,54 @@ func defaultCallbackURLFor(s string) string {
...
+		return
+	})
+	if err != nil {
+		return fmt.Errorf("an unexpected error occurred, please verify your client ID: %w", err)
+	}
+
+	i.Client = client
+	i.QsTypeForClient = quickstartsTypeFor(client.GetAppType())
+
+	quickstarts, err := auth0.GetQuickstarts()
```
Comment (Widcket) [inline internal/cli/quickstarts.go:373]: Should this be inside the spinner func?

**Review by Widcket — CHANGES_REQUESTED** [review]
Looks good, just have a couple of questions.

File: internal/auth0/quickstart.go [hunk 5]
```diff
@@ -1,10 +1,150 @@
...
+
+	if err := os.RemoveAll(downloadPath); err != nil {
+		return err
+	}
+
+	return archiver.Unarchive(tmpFile.Name(), downloadPath)
+}
+
+func GetQuickstarts() (Quickstarts, error) {
+	response, err := http.Get(quickstartsMetaURL)
```
Comment (Widcket) [inline internal/auth0/quickstart.go:96]: Should we use `hettp.NewRequestWithContext()`?

File: internal/auth0/quickstart.go (hunk 1 above)
Comment (sergiught) [reply internal/auth0/quickstart.go:73]: 👌🏻 Yes, what that does is to ensure we have a randomly generated string at the end of that pattern in place of the `*`. 

For example: `auth0-quickstart968365240.zip`. 

From the docs:

```
// CreateTemp creates a new temporary file in the directory dir,
// opens the file for reading and writing, and returns the resulting file.
// The filename is generated by taking pattern and adding a random string to the end.
// If pattern includes a "*", the random string replaces the last "*".
```

File: internal/cli/quickstarts.go (hunk 2 above)
Comment (sergiught) [reply internal/cli/quickstarts.go:137]: We do have `cli.renderer.Warnf()`, however that message needs to go inside the `prompt.Confirm()` func. There doesn't seem to be any function that simply returns a warning string to be used elsewhere. Perhaps I'm missing something? 🤔

File: internal/cli/quickstarts.go (hunk 3 above)
Comment (sergiught) [reply internal/cli/quickstarts.go:348]: 👍🏻  I was thinking that as well, on our code health milestone we could invest into ensuring commands use a similar pattern enforced by an interface.

File: internal/cli/quickstarts.go (hunk 4 above)
Comment (sergiught) [reply internal/cli/quickstarts.go:373]: Good catch! 👍🏻  Definitely.

File: internal/auth0/quickstart.go (hunk 5 above)
Comment (sergiught) [reply internal/auth0/quickstart.go:96]: Sure, that makes sense 👍🏻

Comment (sergiught) [reply internal/auth0/quickstart.go]: [`c06a04f` (#553)](https://github.com/auth0/auth0-cli/pull/553/commits/c06a04f6d41f10032c2078fde1af74910bb0ec3e)

File: internal/cli/quickstarts.go (hunk 4 above)
Comment (sergiught) [reply internal/cli/quickstarts.go]: [`c06a04f` (#553)](https://github.com/auth0/auth0-cli/pull/553/commits/c06a04f6d41f10032c2078fde1af74910bb0ec3e)

File: internal/cli/quickstarts.go (hunk 2 above)
Comment (Widcket) [reply internal/cli/quickstarts.go:137]: That's right, my bad.

**Review by Widcket — APPROVED** [review]

**Pull request merged**